src/evm-block-extractor/src/database/mod.rs

```diff
@@ -92,4 +94,41 @@ pub trait DatabaseClient: Send + Sync {
 
     /// Get earliest block number
     async fn get_earliest_block_number(&self) -> anyhow::Result<u64>;
+
+    /// Delete latest blocks starting with `start_from`, and related transactions.
+    /// Deleted blocks and transactions will be preserved in 'discarded' table with
+    /// the given 'reason' and timestamp.
+    async fn discard_blocks_starting_with(
+        &self,
+        start_from: u64,
+        reason: &str,
```

Comment:
Are we good with reason being a str or should we make it an enum? I haven't checked exactly what its value will be, but str could easily create issues with such a wide domain.

Reply:
I have the similar thought. Since this column intended to provide info for us, when we check the DB, I think string is Ok. If we will decide to use this in some automation, we can re-work it into any enum.